Accept is_pid in is_broadway_name/1 guard

acco opened the pull request.

josevalim commented on lib/broadway.ex:
```diff
@@ -710,7 +710,7 @@ defmodule Broadway do
   """
   @type on_start() :: {:ok, pid()} | :ignore | {:error, {:already_started, pid()} | term()}
-  @type name :: atom() | {:via, module(), term()}
+  @type name :: atom() | pid() | {:via, module(), term()}
```
I don't believe this change is quite right. We now have two names: one used for registration, which can be pids, and one used for access, which cannot be pid. Maybe introduce a:

    @type server :: name() | pid()

Rename is_broadway_name to is_broadway_server and use Broadway.server() where appropriate?

acco commented:
Good call @josevalim just updated. 

It looks like every function outside of stop() requires a name as it uses `:persistent_term` to find the server. So I updated accordingly and added `is_broadway_server()` as a guard for stop()

josevalim commented on lib/broadway.ex:
```diff
@@ -711,6 +711,7 @@ defmodule Broadway do
   """
   @type on_start() :: {:ok, pid()} | :ignore | {:error, {:already_started, pid()} | term()}
   @type name :: atom() | {:via, module(), term()}
+  @type server :: name() | pid()
```
```suggestion
```

josevalim commented on lib/broadway.ex:
```diff
@@ -847,6 +848,9 @@ defmodule Broadway do
   defguardp is_broadway_name(name)
             when is_atom(name) or (is_tuple(name) and tuple_size(name) == 3)
 
+  defguardp is_broadway_server(name_or_pid)
+            when is_broadway_name(name_or_pid) or is_pid(name_or_pid)
+
```
```suggestion
```

josevalim commented on lib/broadway.ex:
```diff
@@ -1226,7 +1230,7 @@ defmodule Broadway do
   @doc since: "0.8.0"
   def stop(broadway, reason \\ :normal, timeout \\ :infinity)
-      when is_broadway_name(broadway) do
+      when is_broadway_server(broadway) do
```
```suggestion
      when is_broadway_name(broadway) or is_pid(broadway) do
```

josevalim commented:
@acco I see, in this case I think a simple `or` guard is enough. Sorry for misleading you. Can you please add a test? Then we should be good, thanks!

acco commented:
@josevalim For sure – just added, look OK?

josevalim commented:
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
